unittests/integration/get_state_tester.cpp

```diff
@@ -68,3 +69,38 @@ CUDAQ_TEST(GetStateTester, checkSimple) {
 
   EXPECT_NEAR(opt_val, 0.0, 1e-3);
 }
+
+CUDAQ_TEST(GetStateTester, checkGetState) {
+
+  auto kernel = []() __qpu__ {
+    cudaq::qubit q, r;
+    h(q);
+    cx(q, r);
+  };
+
+  auto state_object = cudaq::get_state(kernel);
+
+#ifdef CUDAQ_BACKEND_DM
+  // Check that `is_density_matrix` is true.
+  assert(state_object.is_density_matrix() == true);
+  // Can we return the density matrix as an eigen matrix?
+  auto density_matrix = state_object.get_data<Eigen::MatrixXcd>();
+
+  // Is a runtime error thrown if we try to return the density
+  // matrix as a vector?
+  bool error = false;
+  try {
+    auto density_vector = state_object.get_data<Eigen::VectorXcd>();
```

Comment:
You won't need this test anymore when you update to return the pointer, but in the future you might find this macro useful 
```cpp
EXPECT_ANY_THROW({ MY CODE THAT WILL THROW })
```